provider/core/src/resource.rs

```diff
@@ -115,6 +183,7 @@ pub struct ResourceKey {
     // in a compiled binary.
     path: &'static str,
     hash: ResourceKeyHash,
+    metadata: ResourceKeyMetadata,
```

Comment:
Hmm. I'm not sure I see the problem. If you want to pattern match, you're probably better using `&'static str` or `ResourceKeyHash` anyway, because you don't need more than one of those fields in order to identify the key. Also, I think it's reasonable to just say that two ResourceKeys differ if their fallback configs differ.

What would be an alternative? I could make `resource_key!` generate `(ResourceKey, ResourceKeyMetadata)` (or a struct containing those two items as fields) but I don't think that's much better than what I proposed.

Reply:
Note also that I need the fallback metadata inside of BufferProvider, DynProvider, and AnyProvider, which only take two arguments (ResourceKey and DataRequest). At first I had tried putting this on DataRequest, but that is defined at each and every call site, and this is a key-specific config, not a request-specific config. I am not interested in changing BufferProvider, DynProvider, and AnyProvider to take a third argument.

Reply:
> I disagree, conceptually we define equality on resource keys by their path. If we have two keys with the same path but different metadata, we messed up.

An idea I had considered was to add the metadata to the path itself. Something like `plurals/ordinal@1[R]` to indicate that this is a plurals key with region-based fallback. I didn't implement it that way, but now I'm thinking maybe we actually should be explicit like that. For example, if CLDR changes a key from language fallback to region fallback, we likely do actually want to track that as a new key, since the inheritance could result in different data at the same locale.

Reply:
@robertbastian Would you be happy with my proposal above, adding the metadata to the path strings in some form?

Reply:
I wouldn't do that on the API level. However we could *store* it in the string (outside the markers) so we can get cheap reference comparisons.

Reply:
I added the metadata to the path string. I can also remove it from the data model of ResourceKey, but this makes retrieval a bit more expensive. Please let me know how I can resolve this comment thread.